ui/widgets/code_editor.go

```diff
@@ -175,28 +154,108 @@ func (c *CodeEditor) Layout(gtx layout.Context, theme *chapartheme.Theme, hint s
 		layout.Flexed(1, func(gtx layout.Context) layout.Dimensions {
 			return c.border.Layout(gtx, func(gtx layout.Context) layout.Dimensions {
 				return flexH.Layout(gtx,
-					layout.Rigid(func(gtx layout.Context) layout.Dimensions {
-						return listInset.Layout(gtx, func(gtx layout.Context) layout.Dimensions {
-							return material.List(theme.Material(), c.list).Layout(gtx, len(c.lines), func(gtx layout.Context, i int) layout.Dimensions {
-								l := material.Label(theme.Material(), theme.TextSize, fmt.Sprintf("%*d", len(fmt.Sprintf("%d", len(c.lines))), i+1))
-								l.Font.Weight = font.Medium
-								l.Color = theme.TextColor
-								l.TextSize = unit.Sp(14)
-								l.Alignment = text.End
-								return l.Layout(gtx)
-							})
-						})
-					}),
 					layout.Flexed(1, func(gtx layout.Context) layout.Dimensions {
-						return inset4.Layout(gtx, func(gtx layout.Context) layout.Dimensions {
-							ee := material.Editor(theme.Material(), c.editor, hint)
-							ee.TextSize = unit.Sp(14)
-							ee.SelectionColor = theme.TextSelectionColor
-							return ee.Layout(gtx)
+						return layout.Inset{
+							Top:    unit.Dp(4),
+							Bottom: unit.Dp(4),
+							Left:   unit.Dp(8),
+							Right:  unit.Dp(4),
+						}.Layout(gtx, func(gtx layout.Context) layout.Dimensions {
+							editorConf := &giovieweditor.EditorConf{
+								Shaper:          theme.Shaper,
+								TextColor:       theme.Fg,
+								Bg:              theme.Bg,
+								SelectionColor:  theme.TextSelectionColor,
+								TypeFace:        c.font.Font.Typeface,
+								TextSize:        unit.Sp(14),
+								LineHeightScale: 1.2,
+								ColorScheme:     "default",
```

Comment:
ColorScheme can be omitted for now. I planed to implement an internal syntax highlighter for editor of gioview, but seems odd for a general editor. So there's an UpdateTextStyles API to use externally formatted text if needed.